pkg/kube/ready_test.go

```diff
@@ -20,16 +20,212 @@ import (
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &appsv1.Deployment{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady PersistentVolumeClaim",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &corev1.PersistentVolumeClaim{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady Service",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &corev1.Service{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady DaemonSet",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &appsv1.DaemonSet{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady StatefulSet",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &appsv1beta1.StatefulSet{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady ReplicationController",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &corev1.ReplicationController{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "IsReady ReplicaSet",
+			fields: fields{
+				client:        fake.NewSimpleClientset(),
+				log:           func(string, ...interface{}) {},
+				checkJobs:     true,
+				pausedAsReady: false,
+			},
+			args: args{
+				ctx:      context.TODO(),
+				resource: &resource.Info{Object: &extensionsv1beta1.ReplicaSet{}, Name: "foo", Namespace: defaultNamespace},
+			},
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &ReadyChecker{
+				client:        tt.fields.client,
+				log:           tt.fields.log,
```

Comment:
If `log` is always the same, you don't need a variable for it. You can use set it here.

Reply:
This is a stub out I am sure I will refactor this out later.